From f1ef9acecda433a35c403462fc4ee1cd3eb79ea6 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 6 Apr 2016 13:25:56 +0300
Subject: [PATCH 1007/1240] fix: spi: update the clock polarity of the SPI of
 CPHA, and CPOL.

- The SPI device run in 2 modes:
	CPOL=0, CPHA=0
	CPOL=1, CPOL=1
- For these two modes, input data is latched in on the rising edge of Serial Clock (C),
  and output data is available from the falling edge of Serial Clock (C).
- The difference between the two modes, is the clock  polarity when the
  bus master is in standby mode and not transferring data
- Before this change the CPHA and CPOL are not aligned to one another.
- This patch update the CPHA and CPOL parameter to enable/disable them by FDT parameter

Change-Id: I8864a769a59e222d8bca025a0cddec35e752a02e
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28825
Reviewed-by: Haim Boot <hayim@marvell.com>
Tested-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/apn-806-a0.dtsi               | 4 ++--
 arch/arm/dts/apn-806-z1.dtsi               | 4 ++--
 arch/arm/dts/armada-385-db.dts             | 4 ++--
 arch/arm/dts/cp110.dtsi                    | 8 ++++----
 doc/device-tree-bindings/spi/mvebu-spi.txt | 5 +++--
 drivers/spi/mvebu_spi.c                    | 9 +++++----
 6 files changed, 18 insertions(+), 16 deletions(-)

diff --git a/arch/arm/dts/apn-806-a0.dtsi b/arch/arm/dts/apn-806-a0.dtsi
index e298614..6f61f3a 100644
--- a/arch/arm/dts/apn-806-a0.dtsi
+++ b/arch/arm/dts/apn-806-a0.dtsi
@@ -61,8 +61,8 @@
 				reg = <0x510600 0x50>;
 				spi-max-frequency = <10000000>;
 				clock = <&tclk>;
-				cpol-cs-bitmap = <1>; /* bit i is set if the CPOL of
-							 CS-i is enabled or not */
+				cpol-cpha-cs-bitmap = <1>; /* bit i is set if the CPOL and CPHA
+							      of CS-i is enabled or not */
 				status = "disable";
 			};
 
diff --git a/arch/arm/dts/apn-806-z1.dtsi b/arch/arm/dts/apn-806-z1.dtsi
index dc71637..7269902 100644
--- a/arch/arm/dts/apn-806-z1.dtsi
+++ b/arch/arm/dts/apn-806-z1.dtsi
@@ -65,8 +65,8 @@
 				reg = <0x510600 0x50>;
 				spi-max-frequency = <10000000>;
 				clock = <&tclk>;
-				cpol-cs-bitmap = <1>; /* bit i is set if the CPOL of
-							 CS-i is enabled or not */
+				cpol-cpha-cs-bitmap = <1>; /* bit i is set if the CPOL and CPHA
+							      of CS-i is enabled or not */
 				status = "disable";
 			};
 
diff --git a/arch/arm/dts/armada-385-db.dts b/arch/arm/dts/armada-385-db.dts
index 53a2bce..c03d468 100644
--- a/arch/arm/dts/armada-385-db.dts
+++ b/arch/arm/dts/armada-385-db.dts
@@ -79,8 +79,8 @@
 				spi-max-frequency = <20971520>;
 				clock = <&tclk>;
 				status = "okay";
-				cpol-cs-bitmap = <1>; /*bit i is set if the
-							cs-i SCK idles high */
+				cpol-cpha-cs-bitmap = <1>; /* bit i is set if the CPOL and CPHA
+							      of CS-i is enabled or not */
 			};
 
 			spi1: spi@10680 {
diff --git a/arch/arm/dts/cp110.dtsi b/arch/arm/dts/cp110.dtsi
index 19b643a..b2fb765 100644
--- a/arch/arm/dts/cp110.dtsi
+++ b/arch/arm/dts/cp110.dtsi
@@ -33,8 +33,8 @@ spi0@700600 {
 	reg = <0x700600 0x50>;
 	spi-max-frequency = <50000000>;
 	clock = <&refclk>;
-	cpol-cs-bitmap = <0>; /* bit i is set if the CPOL of
-				CS-i is enabled or not */
+	cpol-cpha-cs-bitmap = <0>; /* bit i is set if the CPOL and CPHA
+				      of CS-i is enabled or not */
 	status = "disable";
 };
 
@@ -43,8 +43,8 @@ spi1@700680 {
 	reg = <0x700680 0x50>;
 	spi-max-frequency = <50000000>;
 	clock = <&refclk>;
-	cpol-cs-bitmap = <0>; /* bit i is set if the CPOL of
-				CS-i is enabled or not */
+	cpol-cpha-cs-bitmap = <0>; /* bit i is set if the CPOL and CPHA
+				      of CS-i is enabled or not */
 	status = "disable";
 };
 i2c@701000 { /* i2c0 */
diff --git a/doc/device-tree-bindings/spi/mvebu-spi.txt b/doc/device-tree-bindings/spi/mvebu-spi.txt
index 02aa153..bbe3ca4 100644
--- a/doc/device-tree-bindings/spi/mvebu-spi.txt
+++ b/doc/device-tree-bindings/spi/mvebu-spi.txt
@@ -16,8 +16,9 @@ Board specific:
 	- spi-max-frequency
 		This property is used to calculate the SPI max bus frequency.
 		The value is in Bytes.
-	- cpol-cs-bitmap
+	- cpol-cpha-cs-bitmap
 		This property is bitmap that defines whether the SCK idles high for chip-select i by enabling bit i.
+		This one of SPI modes (to idle in high for, for exmaple in Network over SPI)
 	- status
 		Set if the SPI interface is enabled/disabled.
 
@@ -30,5 +31,5 @@ Example: For SPI-0 node:
 		spi-max-frequency = <20971520>;
 		clock = <&tclk>;
 		status = "okay";
-		cpol-cs-bitmap = <1>; /*bit i is set if the cs-i SCK idles high */
+		cpol-cpha-cs-bitmap = <1>; /*bit i is set if the cs-i SCK idles high */
 	};
diff --git a/drivers/spi/mvebu_spi.c b/drivers/spi/mvebu_spi.c
index 2587a82..4fd35cb 100644
--- a/drivers/spi/mvebu_spi.c
+++ b/drivers/spi/mvebu_spi.c
@@ -153,7 +153,7 @@ struct spi_slave *spi_setup_slave(unsigned int bus, unsigned int cs,
 	struct spi_slave *slave;
 	u32 ctrl_reg, clock = -1;
 	int node_list[CONFIG_MAX_SPI_NUM], node;
-	u32 i, count, cpol = -1;
+	u32 i, count, polarity = -1;
 
 	count = fdtdec_find_aliases_for_id(gd->fdt_blob, "spi",
 			COMPAT_MVEBU_SPI, node_list, CONFIG_MAX_SPI_NUM);
@@ -169,7 +169,7 @@ struct spi_slave *spi_setup_slave(unsigned int bus, unsigned int cs,
 			spi_bus.base_reg = fdt_get_regs_offs(gd->fdt_blob, node, "reg");
 			spi_bus.max_freq = fdtdec_get_int(gd->fdt_blob, node, "spi-max-frequency", CONFIG_MIN_SPI_CLK);
 			clock = soc_clock_get(gd->fdt_blob, node);
-			cpol = (1 << cs) & fdtdec_get_int(gd->fdt_blob, node, "cpol-cs-bitmap", 0);
+			polarity = (1 << cs) & fdtdec_get_int(gd->fdt_blob, node, "cpol-cpha-cs-bitmap", 0);
 		}
 	}
 
@@ -199,9 +199,10 @@ struct spi_slave *spi_setup_slave(unsigned int bus, unsigned int cs,
 	/* Set the SPI interface parameters */
 	ctrl_reg = readl(spi_bus.base_reg + SPI_IF_CONFIG_REG);
 	ctrl_reg &= ~(SPI_CPOL_MASK | SPI_CPHA_MASK | SPI_TXLSBF_MASK | SPI_RXLSBF_MASK);
-	if (cpol)
+	if (polarity) {
 		ctrl_reg |= SPI_CPOL_MASK;
-	ctrl_reg |= SPI_CPHA_MASK;
+		ctrl_reg |= SPI_CPHA_MASK;
+	}
 	writel(ctrl_reg, spi_bus.base_reg + SPI_IF_CONFIG_REG);
 
 	return slave;
-- 
1.9.1

